multidim-interop/src/compose-runner.ts

```diff
@@ -43,14 +44,11 @@ export async function run(namespace: string, compose: ComposeSpecification, opts
     }
 
     try {
-        const timeoutSecs = 3 * 60
-        let timeoutId
-        const { stdout, stderr } =
-            (await Promise.race([
-                exec(`docker compose -f ${path.join(dir, "compose.yaml")} up ${upFlags.join(" ")}`),
-                // Timeout - uses any type because this will only reject the promise.
-                new Promise<any>((resolve, reject) => { timeoutId = setTimeout(() => reject("Timeout"), 1000 * timeoutSecs) })
-            ]))
+        const timeoutId = setTimeout(() => controller.abort(), 1000 * timeoutSecs)
+
+        const controller = new AbortController();
+        const { signal } = controller;
+        const { stdout, stderr } = await exec(`docker compose -f ${path.join(dir, "compose.yaml")} up ${upFlags.join(" ")}`, { signal })
```

Comment:
If `exec` needs an object that has a `signal` property, can't you just pass `controller` directly?

Reply:
This avoids passing in other things that are in the controller. This is also how you do it per the docs: https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback